typescript/packages/wallets/zilliqa/src/ZilliqaWalletClient.ts

```diff
@@ -0,0 +1,80 @@
+// import { type EVMReadRequest, type EVMTransaction, type EVMTypedData, EVMWalletClient } from "@goat-sdk/wallet-evm";
+import { Balance, Chain, Signature, WalletClientBase } from "@goat-sdk/core";
+import { ViemEVMWalletClient } from "@goat-sdk/wallet-viem";
+import { ViemOptions } from "@goat-sdk/wallet-viem";
+import { Account, Wallet } from "@zilliqa-js/account";
+import { Zilliqa } from "@zilliqa-js/zilliqa";
+import { type WalletClient as ViemWalletClient } from "viem";
+
+// Zilliqa has two APIs - EVM and native. The native API is provided by the Zilliqa object. For the EVM API,
+// we thunk down to viem.
+//
+// We need to encapsulate, rather than extend ViemEVMWalletClient so that we can report our chain type correctly.
+// One day it would be nice to add the ability to use ZilPay client integration, but those libraries don't exist yet,
+// so we do this instead.
+export class ZilliqaWalletClient extends WalletClientBase {
```

Comment:
Not familiar with the Zilliqa environment but if there are other implementations of Zilliqa wallets (e.g custodial wallets, smart wallets, etc.) it might be worth creating this as an abstract class and then have a specific implementation (e.g `ViemZilliqaWalletClient`) that handles tx with viem (or zilliqa js).

That would also mean that instead of exposing the viem and zilliqa objects you would have high level functions like sendTransaction or signMessage and each specific implementation would handle them in their own way (e.g Zilliqa implementation would use Zilliqa, Viem would use viem, custodial wallets would use a Rest API, etc.)

It makes sure your wallet design scales for more wallet types and you don't need to change any plugin to make it work for new ones coming

Reply:
I've pulled out as much of it as I think is warranted at this point; this implementation supports a private-key-coded wallet. There are browser integrated wallets that we could support, but abstracting them out would involve building an abstraction for native Zilliqa transactions and completion polling which I'd prefer not to have to stop and build here - however, this framework is half-way there and when we do add support for them, there should hopefully not be much to pull in.